engine/src/org/pentaho/di/trans/steps/mergerows/MergeRowsMeta.java

```diff
@@ -334,6 +335,35 @@ public void check( List<CheckResultInterface> remarks, TransMeta transMeta, Step
           PKG, "MergeRowsMeta.CheckResult.OneSourceStepMissing" ), stepMeta );
       remarks.add( cr );
     }
+
+    RowMetaInterface referenceRowMeta = null;
+    RowMetaInterface compareRowMeta = null;
+    try {
+      referenceRowMeta = transMeta.getPrevStepFields( referenceStream.getStepname() );
+      compareRowMeta = transMeta.getPrevStepFields( compareStream.getStepname() );
+    } catch ( KettleStepException kse ) {
+      // investigate causes
```

Comment:
Swallowed exception here - in the validation, shouldn't that exception be captured and provided as a check result?